crates/emitter/src/forward_jump_emitter.rs

```diff
@@ -0,0 +1,73 @@
+use super::emitter::{BytecodeOffset, EmitError};
+use crate::ast_emitter::AstEmitter;
+
+#[derive(Debug)]
+pub enum JumpKind {
+    Coalesce,
+    LogicalAnd,
+    LogicalOr,
+    IfEq,
+    Goto,
+}
+
+#[derive(Debug)]
+#[must_use]
+pub struct JumpPatchEmitter {
+    pub offset: BytecodeOffset,
+}
+impl JumpPatchEmitter {
+    pub fn emit(self, emitter: &mut AstEmitter) -> Result<(), EmitError> {
+        emitter.emit.patch_jump_target(vec![self.offset]);
+        emitter.emit.jump_target();
```

Comment:
if multiple jumps arrive at same point, only single `JumpTarget` should be emitted
https://searchfox.org/mozilla-central/rev/49ed791eec93335abfe6c2880f84c324e73e47e6/js/src/frontend/BytecodeEmitter.cpp#371-377

I think that can be deferred to the separate issue (given currently there's no code that tracks last target).
can you add `TODO` or `FIXME` comment, with corresponding issue filed?